crates/ra_hir/src/nameres.rs

```diff
@@ -434,6 +434,14 @@ impl ItemMap {
         self.resolve_path_fp(db, original_module, path).0
     }
 
+    pub(crate) fn resolve_name_in_module(&self, module: Module, name: &Name) -> PerNs<ModuleDef> {
+        let from_scope = self[module.module_id].items.get(name).map_or(PerNs::none(), |it| it.def);
+        let from_extern_prelude =
+            self.extern_prelude.get(name).map_or(PerNs::none(), |&it| PerNs::types(it));
+
+        from_scope.combine(from_extern_prelude)
```

Comment:
maybe there should be a better name than `combine` that makes clear what shadows what :smile:

Reply:
`.or`?